### DISCUSSION TITLE
Display future repeating appointment?

### DISCUSSION THREAD

**Discussion by s25765** [mikeberger/borg_calendar#160] [Ideas]
Hi,
BorgCalendar is a great program, I use it to store my todo list. Thank you for developing and maintaining this wonderful application!
However, it doesn't not display my future repeating appointments. Here is one example:

My passport expires on 2025-08-12, I want to be reminded 6 months ahead of time and be reminded weekly,
 so I used Month View to create a repeating appointment by double-clicking the box of February 12th, 2025, and set Recurrence Frequency to weekly, and clicked on the Repeat Forever radio button, then Save & Close. I expected this new appointment will show up on my calendar, but unfortunately it did not.

I looked at the source code and saw the "2 years" limitation, file AppointmentModel.java, around line 230.  Code is:

     if (cyear > curyr + 2)
        break;

I assume getting rid of these 2 lines of code will fix my problem, but I am afraid of doing so because the comment says "an appt repeating 9999 times could kill BORG."

So instead I changed the code a bit on my local PC to temporarily fix my problem. Here is my code:

    int apptYear = cal.get(Calendar.YEAR);
   // ...
   if (cyear > curyr + 2 && cyear > apptYear + 2)
    break;

My code changed the "2 years" rule a little bit, now my future repeating appointments are displayed.
(I can attach the source file if needed.)

Question - What else can I do to make future repeating appointments be displayed?

Thanks!
Steve

**Comment by mikeberger** [comment]
Hi Steve,

I agree that the 2 year limit seems unnecessary on today's computers. I had to look back quite far to find out why there is a 2 year limit and why there was fear of BORG dying.

This code was ported from a C++ version of BORG that I wrote in the early 1990's. It ran on an Intel 486 PC with 4 MB of memory. Repeating an appointment 9999 times actually did kill BORG or at least made it too slow to use.

From the C++ code:


```
// Maximum time in the future to add a repeat. Limits repeats shown
// to about 2 years. This means that large repeat #s are now equivalent
// to "forever" unless you look > 2 years ahead
#define MAX_FUTURE_REPEAT	60*60*24*366*2
#define FOREVER			9999 	// "infinite" number of repeats

...

 if( (*itime - now ) > MAX_FUTURE_REPEAT )
              break;
```
I can just copy your code from this discussion. I'd probably change the 2's to 10's as well.
If you want to submit a pull request, that is fine too. Let me know.

The only way to see your repeats without changing the code would be to wait until 2023 and then they'd show up. Single appointments will also show up any time.

BR,
Mike

**Comment by s25765** [comment]
Thank you for all the detailed info and glad to hear the history.
I like your idea to change 2's to 10's, because a renew of passport expires in 10 years, and a renew of driver's license expires in 5 years.

I used sqlite browser and looked at tables to make sure my new appointment was really stored, without looking at tables I would be quite uncomfortable (was it really store? did I forget to save?) because UI doesn't show it.

Thank you very much! Hope newer releases can show future repeating appointments, and maybe also show current ones with longer years limit.

Thanks!
Steve
